### ISSUE TITLE
Allow users to create sub-interpreters with different site-import settings to the main interpreter

### ISSUE THREAD

**Issue by tonybaloney** [python/cpython#112434]
# Feature or enhancement

### Proposal:

Currently, sub interpreters will inherit the site-import setting from the main interpreter via the `_PyConfig_Copy` call. 

Users can use `-S` to disable site-import for the main and all sub interpreters, but there is no way to create a sub interpreter with site-import disabled. When sub-interpreters are used as a mechanism for parallel computation and don't require site-imports it would be desirable to skip site-import similarly to how `-S` is used for the main process.  

With the default Python 3.13 site.py on macOS, this change cuts interpreter creation by about a third.

```console
$ ./python.exe -m timeit -s "import _xxsubinterpreters as interpreters" -n 100 "interpreters.create(site=False)"
100 loops, best of 5: 19.1 msec per loop
$ ./python.exe -m timeit -s "import _xxsubinterpreters as interpreters" -n 100 "interpreters.create()"          
100 loops, best of 5: 31.1 msec per loop
```

Adding the flag as a parameter to `interpreters.create()` means users can override the site-import flag for the sub interpreter:

```python
import sys

print("sys.flags.no_site", sys.flags.no_site)

import _xxsubinterpreters as interpreters

interp1 = interpreters.create(site=True)
interpreters.run_string(interp1, "import sys; print('site=True', 'sys.flags.no_site', sys.flags.no_site)")

interp2 = interpreters.create(site=False)
interpreters.run_string(interp2, "import sys; print('site=False', 'sys.flags.no_site', sys.flags.no_site)")
```

The sub interpreter will inherit the runtime settings from the main interpreter.

```console
$ ./python.exe test_nosite.py
sys.flags.no_site 0
site=True sys.flags.no_site 0
site=False sys.flags.no_site 1
$ ./python.exe -S test_nosite.py
sys.flags.no_site 1
site=True sys.flags.no_site 0
site=False sys.flags.no_site 1
```

### Has this already been discussed elsewhere?

No response given

### Linked PRs
* gh-112435

**Comment by ericsnowcurrently** [comment]
How feasible would it be to copy the result of site-import from the main interpreter in all subsequent subinterpreters, such that the site module does not need to be imported for each?

**Comment by tonybaloney** [comment]
> opy the result of site-import from the main interpreter in all subsequent subinterpreters, such that the site module does not need to be imported for each?

I don't think that would be possible. `site.py` is mostly side-effects. It writes attributes to the builtin module, changes properties in the sys module. It also optionally runs the user-defined site customize module. 
The most important part is installation of site-packages and user site-packages into `sys.path`.

**Comment by tonybaloney** [comment]
@pablogsal I listened to a whole hour of you and @ambv discussing `site.py` on the podcast. What's your opinion on whether this would be possible?

**Comment by pablogsal** [comment]
> @pablogsal I listened to a whole hour of you and @ambv discussing `site.py` on the podcast. What's your opinion on whether this would be possible?

I think it may be possible, but it has to be probably an opt-in feature as the side effects in `site.py` enables more than just interactive mode and we also don't know if any subinterpreter is going to rely on that.
